controllers/vspheremachine_controller.go

```diff
@@ -242,6 +242,21 @@ func (r *machineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		return reconcile.Result{}, nil
 	}
 
+	// If the VSphereCluster has been previously set as an ownerReference remove it. This may have been set in an older
+	// version of CAPV to prevent VSphereMachines from being orphaned, but is no longer needed.
+	// For more info see: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/2054
+	// TODO (killianmuldoon): Decide when this can be removed.
```

Comment:
Should we remove this comment or code? :-)

Reply:
This was meant to be a bit longer term - i.e. this is migration code and eventually when all existing Clusters have passed through this migration we shouldn't need it any more. I've updated this to be a bit clearer, without asserting when it should be removed.